contract/tests/simulation_test.rs

```diff
@@ -263,209 +281,67 @@ fn alice_adds_liquidity_carol_swaps() {
-    // (At the time of this writing, the error message for an account attempting to transfer tokens to itself would be:
-    // "The new owner should be different from the current owner"
-    // This demonstrates a limitation in simulation testing at the moment. Please see the README for more information on practical debugging steps.
-
-    // Now that we've finished demonstrating that limitation, we'll make the call with the correct
-
-    // Confirm that the simulation account has zero fungible tokens
-    let fungible_tokens: U128 = near_view(
-        &r,
-        &NEP21_ACC.into(),
-        "get_balance",
-        &json!({
-            "owner_id": CLP_ACC
-        })
+    call(
+        &mut ctx.r,
+        &ctx.carol,
+        &ctx.nep21_1,
+        "inc_allowance",
+        format!(
+            r#"{{
+            "escrow_account_id": "{}",
+            "amount": "{}"
+        }}"#,
+            CLP_ACC, carol_tok_from_max_amount
+        ),
+        NEP21_STORAGE_DEPOSIT, //refundable, required if the nep21-contract needs more storage
     );
 
-    assert_eq!(fungible_tokens.clone().0, 0);
-
-    // Give 50 fungible tokens to simulation account
+    println!("nep21-2 {:?}", get_pool_info(&ctx.r, &NEP21_ACC2));
 
-    near_call(&mut r,
-              &fungible_token,
-              &NEP21_ACC,
-              "transfer",
-              &serde_json::to_vec(&json!({
-            "new_owner_id": CLP_ACC,
-            "amount": "50",
-        }),).unwrap(),
-              U64(MAX_GAS),
-              36_500_000_000_000_000_000_000
-    ).unwrap();
-
-    // Now transfer one of those fungible tokens to Alice
-
-    let will_succeed = near_call(&mut r,
-        &simulation_example,
-        &CLP_ACC,
-        "send_token_if_counter_even",
-        &serde_json::to_vec(&json!({
-            "new_num": alice_counter.clone(),
-            "token_account": NEP21_ACC,
-            "recipient_account": ALICE_ACC,
-        }),).unwrap(),
-        U64(MAX_GAS),
-        0
-    ).unwrap();
-
-    println!("Log(s) {:?}", will_succeed.logs);
-    // Make sure it was successful
-    assert_eq!(will_succeed.status, ExecutionStatus::SuccessValue(vec![]));
-
-    alice_tokens = near_view(
-        &r,
-        &NEP21_ACC.into(),
-        "get_balance",
-        &json!({
-            "owner_id": ALICE_ACC
-        })
+    //-- swap_tokens_exact_out
+    call(
+        &mut ctx.r,
+        &ctx.carol,
+        &ctx.clp,
+        "swap_tokens_exact_out",
+        format!(
+            r#"{{
+                "from": "{from}",
+                "to": "{to}",
+                "tokens_to": {tokto},
+                "max_tokens_from": {max_tok_from}
+                }}"#,
+            from = &ctx.nep21_1.account_id(),
+            to = &ctx.nep21_2.account_id(),
+            tokto = 200*NDENOM,
+            max_tok_from = carol_tok_from_max_amount,
+        ),
```

Comment:
this won't work. Please see changes here: https://github.com/robert-zaremba/near-clp/commit/74421b8cb98124ab29c1a62e48f6811d24fb105d#diff-6a1bd9ac7985bc3aaacf14d28dfc5cf2
parameter names changed.